mhkit/utils/type_handling.py

```diff
@@ -1,3 +1,18 @@
+"""
+This module provides utility functions for converting various data types
+to xarray structures such as xarray.DataArray and xarray.Dataset. It also
+includes functions for handling nested dictionaries containing pandas 
+DataFrames by converting them to xarray Datasets.
+
+Functions:
+----------
+- to_numeric_array: Converts input data to a numeric NumPy array.
+- convert_to_dataset: Converts pandas or xarray data structures to xarray.Dataset.
+- convert_to_dataarray: Converts various data types to xarray.DataArray.
+- convert_nested_dict_and_pandas: Recursively converts pandas DataFrames 
+  in nested dictionaries to xarray Datasets.
+"""
```

Comment:
Must have module docstrings

Reply:
My guess is there is a way but I would need to look into it.

If we were going to do that though we would need it to be consistent across all module files.

Reply:
fair. I've actually been updating the API documentation just this morning along with other doc fixes and think I've found away to do this. I'll address that elsewhere in the future with other doc improvements